t/RPC/multiple-rpcs.t

```diff
@@ -88,17 +101,26 @@ subtest 'RPCs should not block each others in different services, same Myriad in
         # if one service's RPC doesn't have messages it should not block the others
 
         for my $i (0..10) {
-            await (fmap {
-                my ($service, $rpc) = shift->%*;
-                $myriad->rpc_client->call_rpc($service, $rpc);
-            } foreach => [
-                {'service.rpc' => 'echo'}, {'service.rpc' => 'ping'},
-                {'another.rpc' => 'zero'}, {'another.rpc' => 'five'},
+            await fmap_void(async sub {
+                my ($service, $rpc, $args, $res) = shift->@*;
+                my $response = await Future->needs_any(
+                    $myriad->rpc_client->call_rpc($service, $rpc, %$args),
+                    $myriad->loop->timeout_future(after => 1)
+                );
+                is_deeply $response, $res, "Matching response $service:$rpc";
+            }, foreach => [
+                ['service.rpc' => 'echo'   , { hi => 'echo' }    , { response => { hi => 'echo' } } ],
+                ['service.rpc' => 'reverse', { v => 'reverseme' }, { response => { reversed => 'emesrever' } } ],
+                ['another.rpc' => 'double' , { v => 4 }          , { response => 8 } ],
+                ['another.rpc' => 'five'   , {}                  , { response => 5 } ],
             ], concurrent => 6);
+            # Calling ping RPC here where it return time is inefficient as we might go to the next second.
         }
         my $done_time = time;
 
-        is($done_time - $start_time < 1, 1, 'RPCs are not blocking each others');
+        # This will cause it to fail sometimes, as it may need more than 1 second to finish.
+        # so to be more realistic, make it no more than 2
+        cmp_ok $done_time - $start_time, '<=', 2, 'RPCs are not blocking each others';
```

Comment:
that's a source of worries, the test runs against the memory transport it should take milliseconds if our setup is correct

Reply:
remember we are using plain `time` which is not that accurate either